packages/spright-components/src/chat/specs/README.md

```diff
@@ -76,17 +69,39 @@ The component also contains the following features:
 
 1. Lays out messages vertically based on their order.
 1. Displays a vertical scrollbar if there are more messages than fit in the height allocated to the conversation.
+1. Includes a slot to place an input component below the messages.
 1. Only appearance of its own is to set a background color.
 
+#### Chat input
+
+1. Accepts text input in a text area
+    - the text area height is a single line initially but grows its height to fit the entered text up to its max-height
+    - the text area has configurable placeholder text
+1. Includes a "Send" button for the user to submit the current input text
+    - fires an event containing the current input content
+    - pressing Enter while the text area has focus will behave the same as clicking "Send"
+    - pressing Shift-Enter will create a newline
+    - the button is disabled and Enter has no effect if the input text and attachments slot are both empty or if the client has set an attribute to explicitly disable it
```

Comment:
Quickly tested Teams mobile vs Teams desktop for multiline and the behavior was:
On desktop pressing enter sent immediately, had to use shift+enter for multiline
On mobile pressing enter added a new line, had to use the send button to send a message

Dunno if we have UX guidance around that, should just do what Teams does, but should capture that somewhere otherwise mobile won't really have a way to do multiline I can think of.

Reply:
Good point, I'll add a note to the mobile section and a task to the issue tracking this work.

Any ideas how to detect that the input came from a mobile keyboard? The only approaches I've found online are to look at user agent or to detect using a heuristic like whether the viewport size has recently changed (presumably due to an onscreen keyboard). Both of those seem like heuristics with less than perfect reliability. Some day maybe [virtual keyboard API](https://developer.mozilla.org/en-US/docs/Web/API/VirtualKeyboard_API) could help, but it's not widely supported.

Reply:
> Any ideas how to detect that the input came from a mobile keyboard? The only approaches I've found online are to look at user agent or to detect using a heuristic like whether the viewport size has recently changed (presumably due to an onscreen keyboard).

Yea not sure off the top. What if they were more aligned? Enter always adds a newline and the button submits, desktop gets an extra shortcut for the submit button (shift + enter).

Reply:
Having stewed on it more I agree with the idea that short quick messages is probably the general workflow. If that's the case I dunno why we would have a different default on mobile. Maybe there is a a well-established UX trend that enter is not commonly used on mobile to submit?

The heuristics based on viewport are also not super promising for mobile detection given our current blazor usage patterns of small / narrow viewports.

Maybe we can move the newline mobile behavior to open questions with the suggestion of aligning with Teams behavior instead of a decision? We might also make it a configuration of the chat-input and let applications decide instead of trying to guess a heuristic in nimble (or have a default auto and make it overridable or something).

Reply:
I updated the description in the Mobile section to describe the challenge and say that we won't solve it now. Also captured a summary of this discussion. I didn't capture it as an open issue in the document since we have a few other similar conscious scoping gaps that aren't listed there either. But it is captured in #2610.